### PR TITLE
WT-10516 Clean up outdated FIXME comments

### PR DESCRIPTION
**Pull request by ajmorton**
Clean up of a handful of FIXMEs where the relevant ticket has been closed. I've added comments for each explaining the change.

### CODE REVIEW

File: src/conn/conn_tiered.c [hunk 1]
```diff
@@ -64,11 +64,6 @@ __tier_storage_remove_local(WT_SESSION_IMPL *session)
               session, WT_VERB_TIERED, "REMOVE_LOCAL: %s in USE, queue again", object);
             WT_STAT_CONN_INCR(session, local_objects_inuse);
             /*
-             * FIXME-WT-7470: If the object we want to remove is in use this is the place to call
```
Comment (ajmorton) [inline src/conn/conn_tiered.c:67]: This is no longer relevant as the `WT_FH` array [has been removed](https://jira.mongodb.org/browse/WT-7470?focusedCommentId=4356010&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4356010).

File: src/docs/arch-fast-truncate.dox [hunk 2]
```diff
@@ -698,12 +698,9 @@ Accept a flag argument that sets \c WT_READ_TRUNCATE.}
 The cursor-level truncate code lives here.}
 
 @row{bt_debug.c, Hook, in \c __debug_cell_int,
-There should be a hook to print the deletion information for \c WT_CELL_ADDR_DEL cells\,
-but as of this writing it is missing.}
-@row{bt_debug.c, Hook, in \c __debug_ref or possibly \c __debug_ref_state,
-There should be a hook to print the deletion information for deleted and instantiated
-pages\, but as of this writing it is missing.}
-<!-- FIXME-WT-9860: WT-9860 is the ticket about these hooks being missing. -->
+Prints the deletion information for \c WT_CELL_ADDR_DEL cells.}
```
Comment (ajmorton) [inline src/docs/arch-fast-truncate.dox:701]: We missed updating the docs as part of WT-9860 here.

File: src/txn/txn_timestamp.c [hunk 3]
```diff
@@ -743,7 +743,7 @@ __wt_txn_set_prepare_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t prepare_
          * (There is a slight extra wrinkle at the moment, because it is possible for a transaction
          * to prepare and commit and be interacted with before it becomes durable. Currently such
          * transactions _must_ be replayed identically by the application to avoid inconsistency,
-         * or avoided. FIXME-WT-8747: remove this note when WT-8747 fixes this.)
```
Comment (ajmorton) [inline src/txn/txn_timestamp.c:746]: We've decided to [document this behaviour but leave it unchanged](https://jira.mongodb.org/browse/WT-8747?focusedCommentId=4376245&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4376245). As such I've left the comment in place.

File: test/csuite/wt2323_join_visibility/main.c [hunk 4]
```diff
@@ -249,13 +249,9 @@ thread_insert(void *arg)
          */
         key = (int)(__wt_random(&rnd) % N_RECORDS);
         maincur->set_key(maincur, key);
-/* FIXME-WT-6180: disable lower isolation levels. */
```
Comment (ajmorton) [inline test/csuite/wt2323_join_visibility/main.c:252]: From the discussion in WT-6180 lower level isolation levels are now read only. There's some talk in the ticket about leaving this code in place, however it's been sitting behind an `#if 0` for 2 years now and it doesn't look like that will change hence the removal. Happy to discuss this one further though.

File: test/evergreen.yml [hunk 5]
```diff
@@ -5167,9 +5167,6 @@ buildvariants:
     CMAKE_PREFIX_PATH: -DCMAKE_PREFIX_PATH="$(pwd)/TCMALLOC_LIB"
     python_binary: '/opt/mongodbtoolchain/v4/bin/python3'
   tasks:
-  # FIXME-WT-8947: For now run the whole stress test suite like we do on the x86 platform. But later
```
Comment (ajmorton) [inline test/evergreen.yml:5170]: We have decided to [keep all stress tests enabled on ARM](https://jira.mongodb.org/browse/WT-8947?focusedCommentId=4548857&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4548857)

File: test/suite/test_schema01.py [hunk 6]
```diff
@@ -78,11 +78,6 @@ def cursor(self, config=None):
         return self.session.open_cursor(self.tablename, None, config)
 
     def test_populate(self):
-        # We skip testing the tiered storage scenarios as we fail to create
-        # column groups in tiered storage scenarios. We should fix this issue
-        # and then remove the condition to skip tests. FIXME: WT-9048
```
Comment (ajmorton) [inline test/suite/test_schema01.py:83]: WT-9058 is now fixed. I've confirmed this and `test_schema02` are now passing when we remove the `skipTest` logic

File: test/csuite/wt2323_join_visibility/main.c (hunk 4 above)
Comment (etienneptl) [reply test/csuite/wt2323_join_visibility/main.c:252]: @quchenhao, would it be worth it to try what was in the `#if 0`?

File: test/suite/test_schema01.py (hunk 6 above)
Comment (etienneptl) [reply test/suite/test_schema01.py:83]: Did you mean WT-9048?

Comment (ajmorton) [reply test/suite/test_schema01.py:83]: Yep, comment fixed 👍

File: test/csuite/wt2323_join_visibility/main.c (hunk 4 above)
Comment (etienneptl) [reply test/csuite/wt2323_join_visibility/main.c:252]: Had a look at the comments as well and it seems that it can be removed indeed.

**Review by etienneptl — APPROVED** [review]
LGTM! Please make sure to sync the `develop` code beforehand and fix the Evergreen failure(s), thanks!

**Comment by sueloverso** [comment]
@ajmorton Thank you for doing this. I thought we had a script in {{dist}} that would flag someone having a PR and open FIXME of the same ticket number. @etienneptl I vaguely remember you wrote it? A quick look through {{dist}} I didn't see it but I know I remember something. Maybe in Evergreen itself rather than dist?

Anyway, the point of my comment is can we know why did that script not catch the things you're fixing here? Could that gap be fixed?

**Comment by sueloverso** [comment]
@etienneptl thank you for the reminder and explanation. That makes a lot of sense.

**Comment by ajmorton** [comment]
Yep, thanks @etienneptl for the summary.
@sueloverso the script you're thinking of is likely `s_mentions`, which as Etienne mentioned only applies to the ticket a developer is currently working on. That script will look for any mention of the ticket, but only notifies the developer about the mentions and doesn't fail `s_all`. This is because there are cases where we intentionally leave a a relevant comment in the codebase such as [this reference to WT-6778 here](https://github.com/wiredtiger/wiredtiger/blob/ab4c2cc56edd41e9a55856740e05fab2c8108b11/src/txn/txn.c#L1092).

The two cases main cases this new script catches are:
1. A developer misses a warning from `s_mentions` as described above, and 
2. A ticket is closed as `Won't Fix`/`Duplicate` in Jira without an update to its FIXME in the code.

**Comment by ajmorton** [comment]
evergreen merge

**Pull request merged**